### ISSUE TITLE
Unmaintained

### ISSUE THREAD

**Issue by Traviskn** [joltup/rn-fetch-blob#666]
I am no longer employed by the company behind [Joltup](https://github.com/joltup), and I do not believe this library will be well-maintained for community usage. Even during my time working for Joltup, I was not able to dedicate the time necessary to keep up with PRs and questions. Unless something has significantly changed at Joltup, I don't expect issues or PRs to be addressed very quickly going forward.

At this point if there are those in the community who have the time and wherewithal to maintain rn-fetch-blob, I would recommend either forking this library and publishing it under a new name, or perhaps starting a new file transfer library from scratch using this library and other existing solutions as some inspiration. Considering that react native now [supports the Blob API](https://github.com/facebook/react-native/releases/tag/v0.54.0) I think that there is a lot of bloat in rn-fetch-blob that could be stripped out. [Expo File System](https://github.com/expo/expo/tree/master/packages/expo-file-system) continues to be a good alternative, relying on react native core functionality for file uploads, and only bridging native code for file downloads and local file management.

**Comment by LouisJS** [comment]
This is a very important news. There is so many projects that relies on it. We should mention it in the ReadMe and redirects user to Expo File System in Bare workflow. It shouldn't be just mention on an issue that is going to be lost !

**Comment by cristianoccazinsp** [comment]
Can we get yet another fork?  This is really bad news, this library is extremely useful and using Expo just for file system access is overkill (considering all the stuff that has to be installed).

**Comment by dommangetnicolas** [comment]
> Can we get yet another fork? This is really bad news, this library is extremely useful and using Expo just for file system access is overkill (considering all the stuff that has to be installed).

You don't need to use the entire Expo workflow. It is possible to install only certain blocks with [Expo Unimodules](https://docs.expo.io/bare/installing-unimodules/).

**Comment by cristianoccazinsp** [comment]
It still means having a dependency with Expo, and also means having to wait
for Expo to update their libs after each RN upgrade. I mean it's not the
end of the world, and I understand there's an attempt to move more people
into using expo, but it's a dependency I would really like to avoid if
possible.

**Comment by sylvanaar** [comment]
Wow! what awful governance. To take an important and well-used library, and just neglect it. Why take it over in the first place, and why not find a maintainer for it before you just abandon it?

**Comment by skyliwq** [comment]
[react-native-blob-courier ](https://github.com/edeckers/react-native-blob-courier) new open source project no one has ever heard of

**Comment by alpha0010** [comment]
Just started drafting out a new filesystem library: https://github.com/alpha0010/react-native-file-access .  Scope is file operations and download url directly to disk.  Uses Kotlin/Swift.  Low impact on app size (approx. 200 lines native code per platform).  (Note: no "production ready" stamp yet.  Use at your own risk, or wait while I continue testing it.)

**Comment by lumberman** [comment]
issue #666

**Comment by caojianfeng** [comment]
There are so many issues and pull requires . I think we need a group of maintainers.
Can you join my org:https://github.com/orgs/ovotop or found a new one?

I had sent an invite to you.

**Comment by RonRadtke** [comment]
I decided to frok it and started fixing some stuff and integrated some pull requests. Including Support for Android > API 28
You can find it here: [](https://github.com/RonRadtke/rn-fetch-blob)

**Comment by nixoschu** [comment]
I can totally understand if someone drops the maintenance on a repo no matter the cause.
But this repo still has around ~100k downloads a week which makes it very vital to a lot of people.

So what exactly is the go to from here ?
- Not use the package because it's no longer maintained ?
- Still use it until it breaks or hope for another maintainer ?
- Using another repo ? What are the alternatives ?

Using expo or part of the unimodules seems kind of an overkill if you want to download a single file somewhere.

There are a lot of answers with other repos or forks that can work in a same way, but (no front to anyone) who can guarantee support on one of these.

Wouldn't it be a good option to get in touch with e.g. react-native-community (or else) so this repo maybe gets some more "official" support since downloading, uploading and storing files is a very common task for a lot of apps.

Anyway, a statement on how to proceed with this repo for the developers using it would be very much appreciated.

**Comment by cristianoccazinsp** [comment]
I agree on putting this under the react-native-community umbrella to get some additional support. This library is vital to pretty much any RN project and using Expo or Unimodules is a pain in the neck and overkill. 

React Native is great, but having to constantly keep up with dependencies going stale / getting unmaintained is quite an issue nowadays.

**Comment by RonRadtke** [comment]
@nixoschu I forked it and will maintain it there.
So far I added the implementation for windows and support for Android 10, 11. Also I changed it so we get feedback for Android if there is no app to open a file with.
You can find it here: https://github.com/RonRadtke/react-native-blob-util or https://www.npmjs.com/package/react-native-blob-util
The next days I also will update the Readme for new installation hints, etc.
If anyone wants to help maintaining I'm always open for help!

**Comment by flashman2** [comment]
Hey @Traviskn 

This library is forked from original react-native-fetch-blob https://github.com/wkh237/react-native-fetch-blob
In 2017 original library lost support/maintenance
But as everyone can see, in the main page of this library added a note:
---
This repository no longer is the main location of "react-native-fetch-blob".
The owners of this fork have agreed to maintain this package:
https://github.com/joltup/rn-fetch-blob
That means issues and PRs should be posted there.
---

What's the next home for this library? Can you please add the same note to README? There is a few people in the comments who want to maintain this library. Who will be "official" supporter?

**Comment by pke** [comment]
https://github.com/edeckers/react-native-blob-courier it is for me. It's actively developed and improved.

**Comment by wumke** [comment]
Since @RonRadtke forked this project and it is fully compatible I'd suggest putting this link as maintained and compatible substitute in the README of this project: https://www.npmjs.com/package/react-native-blob-util.

I also suggest @RonRadtke updates the readme of his fork such that it mentions that his package is a substitute for https://www.npmjs.com/package/rn-fetch-blob, which is a substitute for https://github.com/wkh237/react-native-fetch-blob.

This way users find their way to a maintained fully compatible fork. 
=> @flashman2 : you can use https://www.npmjs.com/package/react-native-blob-util

@Traviskn : Please make one final update to the readme as described above... (And thanks for picking up and maintaining the initial project up to here!)

**Comment by billnbell** [comment]
Hey, can you add release numbers and put into NPM?

You can find it here: https://github.com/RonRadtke/react-native-blob-util or https://www.npmjs.com/package/react-native-blob-util

**Comment by RonRadtke** [comment]
@wumke I updated the readme of my project as you suggested.
@billnbell for which project do you want release numbers? Mine or this one here

**Comment by flashman2** [comment]
@RonRadtke Thank you! Great job. Glad that your fork is the new official home for this module! :)

**Comment by billnbell** [comment]
Whatever project is taking over, you should have release tags, and publish on NPMJS

**Comment by RonRadtke** [comment]
@billnbell Added a release. It's on NPM for quite a while already

**Comment by kelset** [comment]
Hey @RonRadtke thanks for deciding to take this on yourself, really appreciated!

Do you have a way I can reach out to you directly? I want to see if there are ways to make sure that things will be smoother on this next fork :)

(if you don't want to share your details here please just send me an email at notkelset@kelset.dev 🤗)

**Comment by cristianoccazinsp** [comment]
Hello all.  Had to dig deep to find back this issue on my emails.  Is it possible to update this repo in order to point to the new home of this fork?
